src/include/storage/sql_table.h

```diff
@@ -102,45 +231,59 @@ class SqlTable {
    * @param start_pos iterator to the starting location for the sequential scan
    * @param out_buffer output buffer. The object should already contain projection list information. This buffer is
    *                   always cleared of old values.
+   * @param pr_map the ProjectionMap for the Projected Columns
+   * @param version_num the schema version which the transaction sees
    */
-  void Scan(transaction::TransactionContext *const txn, DataTable::SlotIterator *const start_pos,
-            ProjectedColumns *const out_buffer) const {
-    return table_.data_table->Scan(txn, start_pos, out_buffer);
-  }
+  void Scan(transaction::TransactionContext *txn, SqlTable::SlotIterator *start_pos, ProjectedColumns *out_buffer,
+            const ProjectionMap &pr_map, layout_version_t version_num) const;
 
   /**
-   * @return table's unique identifier
+   * @return the first tuple slot contained in the data table
    */
-  catalog::table_oid_t Oid() const { return oid_; }
+  SlotIterator begin(layout_version_t txn_version) const {
+    // common::SpinLatch::ScopedSpinLatch guard(&tables_latch_);
+    auto ret = SlotIterator(&tables_, txn_version, false);
+    ret.AdvanceOnEndOfDatatable_();
+    return ret;
+  }
 
   /**
-   * @return the first tuple slot contained in the underlying DataTable
+   * Returns one past the last tuple slot contained in the last data table. Note that this is not an accurate number
```

Comment:
Confusing document: What is the `number` it is talking about? Tuple slot or something else?